lib/utils.js

```diff
@@ -215,3 +218,16 @@ export const flattenObjectDeep = obj =>
   );
 
 export const stripHTML = htmlContent => sanitizeHtml(htmlContent, { allowedTags: [], allowedAttributes: {} });
+
+export const checkRequiresAddress = values => {
+  const isSelfHostedCollective = values.payee.id === values.payee.host?.id;
+  return (
+    values.payee &&
+    (![CollectiveType.COLLECTIVE, CollectiveType.EVENT, CollectiveType.PROJECT, CollectiveType.FUND].includes(
+      values.payee.type,
+    ) ||
+      isSelfHostedCollective) &&
+    !values.payee.isInvite &&
+    [expenseTypes.INVOICE, expenseTypes.FUNDING_REQUEST, expenseTypes.GRANT].includes(values.type)
+  );
```

Comment:
I've used some intermediary variables and also changed it to the `CollectiveTypes` that need an address rather than trying to use the negated statement. To me this looks simpler, but let me know.